### PR TITLE
[openstack] Support for tagging workers

### PR DESCRIPTION
**Pull request by iamemilio**
- Tags workers with Name, and openstack cluster id
- Pulls in changes to capo from upstream

### CODE REVIEW

File: pkg/asset/machines/openstack/machines.go [hunk 1]
```diff
@@ -111,6 +111,10 @@ func provider(clusterID string, platform *openstack.Platform, mpool *openstack.M
 				Name: fmt.Sprintf("%s-%s", clusterID, role),
 			},
 		},
+		ServerMetadata: map[string]string{
+			"Name":               clusterID + "-worker",
```
Comment (flaper87) [inline pkg/asset/machines/openstack/machines.go]: Probably better to do `fmt.Sprintf("%s-%s", clusterID, role)`

Comment (iamemilio) [reply pkg/asset/machines/openstack/machines.go]: I actually just did that in my local lol

**Pull request merged**